packages/editor/src/client/panel/navigationEditPanel.tsx

```diff
@@ -0,0 +1,336 @@
+  const [pages, setPages] = useState<PageRefFragment[]>([])
+  const [articles, setArticles] = useState<ArticleRefFragment[]>([])
+
+  const testLinkTypes: Record<string, string>[] = []
+  for (const [propertyValue, propertyKey] of Object.entries(LinkTypes)) {
+    // dump-variable is a work-around to avoid "unused-variable"-error
+    testLinkTypes.push({label: propertyKey, value: propertyKey, dump: propertyValue})
+  }
+
+  const {data, loading: isLoading, error: loadError} = useNavigationQuery({
+    variables: {id: id!},
+    fetchPolicy: 'network-only',
+    skip: id === undefined
+  })
+
+  const {data: pageData, loading: isLoadingPageData, error: pageDataError} = usePageListQuery({
+    variables: {first: 50},
+    fetchPolicy: 'no-cache'
+  })
+
+  const {
+    data: articleData,
+    loading: isLoadingArticleData,
+    error: articleDataError
+  } = useArticleListQuery({
+    variables: {first: 50},
+    fetchPolicy: 'no-cache'
+  })
+
+  const [createNavigation, {loading: isCreating, error: createError}] = useCreateNavigationMutation(
+    {
+      refetchQueries: [getOperationNameFromDocument(NavigationListDocument)]
+    }
+  )
+
+  const [
+    updateNavigation,
+    {loading: isUpdating, error: updateError}
+  ] = useUpdateNavigationMutation()
+
+  const isDisabled =
+    isLoading ||
+    isCreating ||
+    isUpdating ||
+    isLoadingPageData ||
+    loadError !== undefined ||
+    pageDataError !== undefined ||
+    isLoadingArticleData ||
+    articleDataError !== undefined
+
+  const {t} = useTranslation()
+
+  useEffect(() => {
+    if (data?.navigation) {
+      setName(data.navigation.name)
+      setKey(data.navigation.key)
+      setNavigationLinks(
+        data.navigation.links
+          ? data.navigation.links.map(link => {
+              return {
+                id: generateID(),
+                value: {
+                  label: link.label,
+                  type:
+                    link.__typename === 'ArticleNavigationLink'
+                      ? 'Article'
+                      : link.__typename === 'PageNavigationLink'
+                      ? 'Page'
+                      : 'External',
+                  articleID:
+                    link.__typename === 'ArticleNavigationLink' ? link.article?.id : undefined,
+                  pageID: link.__typename === 'PageNavigationLink' ? link.page?.id : undefined,
+                  url: link.__typename === 'ExternalNavigationLink' ? link.url : undefined
+                }
+              }
+            })
+          : []
+      )
+    }
+  }, [data?.navigation])
+
+  useEffect(() => {
+    if (pageData?.pages?.nodes) {
+      setPages(pageData.pages.nodes)
+    }
+  }, [pageData?.pages])
+
+  useEffect(() => {
+    if (articleData?.articles.nodes) {
+      setArticles(articleData.articles.nodes)
+    }
+  }, [articleData?.articles])
+
+  useEffect(() => {
+    const error = loadError?.message ?? createError?.message ?? updateError?.message
```

Comment:
it would be wise if you also show the `articleDataError` and `pageDataError`. Also i would rename them to `articleLoadError` and `pageLoadError`.